app/models/manageiq/providers/redhat/infra_manager/ovirt_services/strategies/v3.rb

```diff
@@ -227,9 +227,24 @@ def method_missing(method_name, *args, &block)
       end
     end
 
-    def find_mac_address_on_network(nics, network, log)
+    def get_mac_address_of_nic_on_requested_vlan(args)
+      network = find_network_in_cluster(args[:value_of_vlan_option], args[:dest_cluster], args[:source_template])
+      return nil if network.nil?
+
+      find_mac_address_on_network(network, args[:destination])
+    end
+
+    def find_network_in_cluster(network_name, dest_cluster, source)
+      network = source.ext_management_system.ovirt_services.cluster_find_network_by_name(dest_cluster.ems_ref, network_name)
```

Comment:
1. I didn't mean you need to move the method to another file. In network.rb invoke **source**.ext_management_system.ovirt_services.get_mac_address_of_nic_on_requested_vlan(args) instead of destination.ext_management_system.ovirt_services.get_mac_address_of_nic_on_requested_vlan(args).
2. 'nics_for_vm(..)' invokes
'vm.with_provider_object' which invokes
'connection_source(..)' which invokes
def connection_manager
    try(:ext_management_system) || try(:manager)
 end
So, IIUC, 'ext_management_system' of the vm is invoked.
So why here it is working and in your case it is not?